internal/provisioner/controller/gateway.go

```diff
@@ -292,6 +292,10 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 					Tolerations:  nodePlacement.Tolerations,
 				}
 			}
+
+			for k, v := range gatewayClassParams.Spec.Envoy.PodAnnotations {
```

Comment:
hm, we don't actually have any coverage of the actual logic in this change which is unfortunate, this would have to be covered by a test in this package

probably important to test that we're appending on top of the existing annotations since that ends up becoming API surface

do we want to prevent overwriting of existing annotations?